### PR TITLE
Add stripe-auth related endpoints docs and dredd tests

### LINKED ISSUE #4376: Stripe-Authorization: Add related endpoints to docs and dredd tests
Parent issue #4009.

### PR DESCRIPTION
**Pull request by enigmaeth**
Fixes #4376.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

- Add docs for stripe related endpoints.
- Remove `events/id/stripe-authorization` from list as one event can have only one stripe auth.

### CODE REVIEW

File: app/api/stripe_authorization.py [hunk 1]
```diff
@@ -31,10 +31,6 @@ def before_create_object(self, data, view_kwargs):
             raise ConflictException({'pointer': '/data/relationships/event'},
                                     "Stripe Authorization already exists for this event")
 
-    def before_get(self, args, kwargs):
-        if not has_access('is_super_admin'):
-            raise ForbiddenException({'source': ''}, "Super Admin Access Required")
-
     schema = StripeAuthorizationSchema
```
Comment (poush) [inline app/api/stripe_authorization.py]: Add POST only method here

File: app/api/stripe_authorization.py [hunk 2]
```diff
@@ -46,38 +42,40 @@ class StripeAuthorizationList(ResourceList):
     Stripe Authorization List Resource
     """
 
-    def query(self, view_kwargs):
```
Comment (poush) [inline app/api/stripe_authorization.py]: Why StripeAuthorizationList is not completely removed?

File: app/api/stripe_authorization.py [hunk 3]
```diff
@@ -46,38 +42,40 @@ class StripeAuthorizationList(ResourceList):
...
     """
     Stripe Authorization Detail Resource by ID
     """
+    def before_get_object(self, view_kwargs):
+        """
+        method to get id of stripe authorization related to an event
+        :param view_kwargs:
+        :return:
+        """
+        if view_kwargs.get('event_identifier'):
```
Comment (poush) [inline app/api/stripe_authorization.py]: You can use event_query for this

File: app/api/stripe_authorization.py (hunk 1 above)
Comment (enigmaeth) [inline app/api/stripe_authorization.py]: Resolved.

File: app/api/stripe_authorization.py (hunk 3 above)
Comment (enigmaeth) [inline app/api/stripe_authorization.py]: I don't think that event_query can be used here to get the detail. Correct me if I am wrong :)

File: app/api/stripe_authorization.py (hunk 2 above)
Comment (enigmaeth) [inline app/api/stripe_authorization.py]: @poush Resolved.

File: app/api/stripe_authorization.py (hunk 3 above)
Comment (poush) [inline app/api/stripe_authorization.py]: ah! My bad 😅

**Comment by shubham-padia** [comment]
@poush Please review

**Comment by shubham-padia** [comment]
@poush Please review

**Comment by enigmaeth** [comment]
@poush Can you please review this now :sweat_smile: ?

**Comment by poush** [comment]
Sorry, 😅 I'm not sure how I missed it.

**Pull request merged**